### ISSUE TITLE
Skipped tasks shown as pending

### ISSUE THREAD

**Issue by larhauga** [tektoncd/dashboard#2293]
# Describe the bug

Pipelines with tasks that are skipped (like due to non master branch run) is observed to have the state of pending even though it has been skipped. This issue came to us from end-users, which did not understand why the task was still pending.

![image](https://user-images.githubusercontent.com/456305/150386900-b30fce89-af87-47d5-9b7b-9a880f083bb7.png)

# Expected behaviour
We expect the status of the skipped task to be `Skipped`; not `Pending`.

# Steps to reproduce the bug
In this case we have a `when` condition, that skipps this task if branch is main or master.
```
...
    - name: promote
      runAfter:
      - mvn-build
      taskRef:
        kind: ClusterTask
        name: # taskname
      when:
      - input: $(params.gitbranchname)
        operator: in
        values:
        - master
        - main
...
```

# Environment details

- Kubernetes Platform: EKS

- Kubernetes or OpenShift version: 1.21

- Install mode (if on OpenShift): no

- Cloud-provider/provisioner: EKS

- Versions:
   - Tekton Dashboard: 0.23.0
   
   - Tekton Pipelines: 0.31.0
   
   - Tekton Triggers: 0.18.0
   
- Install namespaces:
   - Tekton Dashboard: tekton-pipelines
   
   - Tekton Pipelines:
   
   - Tekton Triggers:
   

# Additional Info

**Comment by AlanGreene** [comment]
Thanks for raising this issues @larhauga. We're currently considering a redesign of this page to improve display of a number of things including when expressions, retries, and more.

**Comment by AlanGreene** [comment]
This is something we'll be addressing as part of the PipelineRun details page redesign. Freezing so it doesn't get auto closed.

/lifecycle frozen

**Comment by AlanGreene** [comment]
When we address this we should review https://github.com/tektoncd/pipeline/pull/4829 and TEP-0103. This makes additional useful information available regarding the reason for a task being skipped.

**Comment by AlanGreene** [comment]
Hoping to revisit this soon. Related to https://github.com/tektoncd/dashboard/issues/1445

**Comment by AlanGreene** [comment]
Blocked by https://github.com/tektoncd/dashboard/issues/3493 which in currently is progress. It affects pretty much the entire front-end codebase so we've paused functional changes util that's complete, ETA ~end of month.

Once that's done, this is among the next batch of UI updates planned.

**Comment by AlanGreene** [comment]
#3651 improves the display of skipped tasks, making it clearer to users that these are skipped and not actually pending. The corresponding when expression or other relevant reason is displayed in the taskrun status, and the log container clearly identifies that no logs will be displayed due to the task being skipped.

This will be available in the next nightly release, and included in Dashboard v0.52.0 which is expected in the next week or so.

Further improvements will be made as part of https://github.com/tektoncd/dashboard/issues/2306

**Comment by AlanGreene** [comment]
This is now available in https://github.com/tektoncd/dashboard/releases/tag/v0.52.0
